src/main/java/seedu/address/logic/commands/ActivityCommand.java

```diff
@@ -102,10 +102,11 @@ public CommandResult execute(Model model) throws CommandException {
         }
 
         Activity toAdd = new Activity(title, participantIds.toArray(new Integer[participantIds.size()]));
+        Context newContext = new Context(toAdd);
         model.addActivity(toAdd);
-        model.setContext(new Context(toAdd));
+        model.setContext(newContext);
         model.updateFilteredPersonList(x -> toAdd.getParticipantIds().contains(x.getPrimaryKey()));
-        return new CommandResult(String.format(MESSAGE_SUCCESS, toAdd, successMessage, warningMessage));
+        return new CommandResult(String.format(MESSAGE_SUCCESS, toAdd, successMessage, warningMessage), newContext);
```

Comment:
Works as expected (UI only updates when activity is successfully added).

@daekoon Aside: the activity command does not execute correctly (showing an error message for empty participant argument) when the participant tag `p/`is entered with no value, i.e. in `activity p/ t/Sample activity`.